pkgs/top-level/config.nix

```diff
@@ -131,10 +131,26 @@ let
     checkMeta = mkOption {
       type = types.bool;
       default = false;
-      description = ''
+      description = lib.mdDoc ''
         Whether to check that the `meta` attribute of derivations are correct during evaluation time.
       '';
     };
+
+    packageOverrides = mkOption {
+      default = const {};
+      type = types.overrideFnType;
+      description = lib.mdDoc ''
+        **Not recommended.** Use `overlays` instead!
+      '';
+    };
+
+    perlPackageOverrides = mkOption {
+      default = const {};
+      type = types.overrideFnType;
+      description = lib.mdDoc ''
+        A function that takes the final set of Perl packages and returns an attribute set of packages to add or change.
```

Comment:
In addition, the `perlPackageOverrides` is super weird, in that it doesn't even take the perl packages as an argument, but rather the main `pkgs` package set!

```
❯ nix-instantiate '<nixpkgs>' --arg config '{ perlPackageOverrides = pkgs: builtins.trace (builtins.head (builtins.attrNames pkgs)) {}; }' -A perlPackages.FileFinder
trace: AAAAAASomeThingsFailToEvaluate
trace: AAAAAASomeThingsFailToEvaluate
trace: AAAAAASomeThingsFailToEvaluate
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/sf3bji32b4xw77xjh7w452pj15jq8gh6-perl5.34.1-File-Finder-0.53.drv
```

(this shows that the first attribute in the passed `pkgs` is `pkgs.AAAAAASomeThingsFailToEvaluate`)

Reply:
no worries, I also missed that!

Reply:
Uhm... I obseve the same behavior on my local `master` (dfae1c183f97ba1f89e0a78ce0046a8d9198fd93 currently):

```
❯ ~/Projects/nixpkgs master → nix-instantiate './.' --arg config '{ perlPackageOverrides = pkgs: builtins.trace (builtins.head (builtins.attrNames pkgs)) {}; }' -A perlPackages.FileFinder
trace: AAAAAASomeThingsFailToEvaluate
trace: AAAAAASomeThingsFailToEvaluate
trace: AAAAAASomeThingsFailToEvaluate
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/f9qgccfq4p6n161jlkdf05qikmq4l9jj-perl5.36.0-File-Finder-0.53.drv
```

Am I missing something? oO

Reply:
That's as expected. This just demonstrates that `perlPackageOverrides` gets the `pkgs` as an argument and not `pkgs.perlPackages`, since there's only `pkgs.AAAAASomeThingsFailedToEvaluate` and no `pkgs.perlPackages.AAAAASomeThingsFailedToEvaluate`. Don't mind this specific attribute name, it's not related to this issue.

Reply:
> In addition, the `perlPackageOverrides` is super weird, in that it doesn't even take the perl packages as an argument, but rather the main `pkgs` package set!

It's funny. In a discussion today the inverse came up. Without a way to reference `pkgs`, mere two-function overlays are not good enough for distributing additions/changes to language-specific package sets, because those inevitably tend to need some things from `pkgs`... but which one?
So without a `pkgs` parameter it's not suitable for distributed use, and without a `prev` or `self`, it's not an overlay.
We need both! ... or do we? `self` is actually available in `pkgs`.
But yes, we do, because the location may be ambiguous. Is it `haskellPackages` or `haskell.packages.ghc943`?